### ISSUE TITLE
Different tones on different parts of the text

### ISSUE THREAD

**Issue by porky11** [rhasspy/piper#490]
It would be nice if it was possible to add some specific tones to single words or whole sentences.

For example:
- emphasis
- sarcasm
- whispering

(also all combinations, like whispering sarcasm)

Is this already possible somehow?
I saw espeak generates emphasis markers anyway, but maybe this could be altered manually in some way?

Or I could probably train and use different variants of some voice. But it seems it's not possible to switch voices without causing pauses, even when setting "sentence_silence" to 0. But this would probably be the best workaround so far.

It would still be nice if such a feature existed, preferably without the need of training new voices (if possible).

**Comment by Daburnell112** [comment]
I second this. Some sort of markup language would be nice if it exists.

**Comment by rmcpantoja** [comment]
Hi,
I will work in a pitch conditioning model soon and maybe PR these additions with new updates.

**Comment by nmstoker** [comment]
@porky11 - how could sarcasm or whispering be applied to an output voice without training with relevant voice recordings?

Is there some proces you have in mind that could be applied to the audio to achieve this? My suspicion is that there isn't a viable way to do this (without the audio + training)

**Comment by synesthesiam** [comment]
In my experience, you need audio data for each case (sarcasm, whispering, etc.) and then a multi speaker model needs to be trained with each case being a different "speaker".

This is exactly what the Thorsten emotional voice does (German).

**Comment by ashkitten** [comment]
> I second this. Some sort of markup language would be nice if it exists.

Yes, the standard one is https://www.w3.org/TR/speech-synthesis11/
